meow/user_profile/views.py

```diff
@@ -5,29 +5,112 @@
+        themeOrderedDict = serialized_themes.data
+        return JsonResponse(themeOrderedDict, safe=False, status=200)
+    
+    def post(self, request, id, format=None):
+        user = request.user
+        req_data = request.data
+        new_name =  req_data.get('name', None)
+        serializer = ThemeSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save(author=request.user)
+        else:
+            print(serializer.errors)
+            return JsonResponse(serializer.errors, safe=False, status=400)
+        new_id = Theme.objects.get(name=new_name).pk
+        return JsonResponse(new_id, safe=False, status=200)
+
+    def put(self, request, id, format=None):
+        user=request.user
+        req_data = request.data
+        if(id == 1 or id == 2):
+            return HttpResponse('Default themes cannot be modified', status=400)
+        serializer = ThemeSerializer(Theme.objects.get(pk=id), data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+        else:
+            print(serializer.errors)
+            return JsonResponse(serializer.errors,status=400, safe=False)
+        return JsonResponse(id, safe=False, status=200)
+    
+    def delete(self, request, id, format=None):
+        user=request.user
+        themetodelete = Theme.objects.get(pk=id)
+        delete_name=themetodelete.name
+        if(delete_name == 'Daily Bruin' or delete_name == 'Dark Bruin'):
+            return JsonResponse('Default themes cannot be deleted', safe=False, status=400)
+        if user.selected_theme == Theme.objects.filter(name=delete_name, author=user)[0]:
+            user.selected_theme = Theme.objects.get(name="Daily Bruin")
+            user.save()
+        if themetodelete.user_set.count()==0 and themetodelete.related_users.count()==0:
+            themetodelete.delete()
+        else:
+            themetodelete.author=None
+            themetodelete.save()
+        serialized_theme = ThemeSerializer(user.selected_theme, many=False)
+        themeOrderedDict = serialized_theme.data
+        return JsonResponse(themeOrderedDict, safe=False, status=200)
+        
 
 @api_login_required()
-def themeList(request):
+def additionalthemeList(request):
+    #this functon list out the additional themes
+    user = request.user
     if request.method == "GET":
-        themes = Theme.objects.all()
+        themes = Theme.objects.exclude(author=user).exclude(name__in=["Daily Bruin", "Dark Bruin"]).order_by('-favorite_count')
         serialized_themes = ThemeSerializer(themes, many=True)
         themeOrderedDict = serialized_themes.data
-        return JsonResponse(themeOrderedDict, safe=False)
+        starred_themes_id =[theme.pk for theme in user.starred_themes.all()] 
+        return JsonResponse({'additionalThemes': themeOrderedDict, 'starredThemesId': starred_themes_id }, safe=False, status=200)
+
+@api_login_required()
```

Comment:
It is better to use Django rest framework's decorators. 
```
from rest_framework.decorators import api_view, permission_classes


@api_view(['POST', 'PUT'])
@permission_classes([IsAuthenticated])
def themeStar(request):
```

If you don't use django rest framework decorators or class based views, you can't use `request.data`

Reply:
fixed